From ff66709a8176ccd76e28d58001e8e52a609035ef Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?R=C3=A9mi=20Bernon?= <rbernon@codeweavers.com>
Date: Tue, 12 Nov 2019 18:38:45 +0100
Subject: [PATCH 09/16] user32/tests: Tweak test_PeekMessage expectations for
 w1064v1809.

This sometimes fails on the testbot, usually on the non-us VMs.

  https://testbot.winehq.org/JobDetails.pl?Key=59888#k113
---
 dlls/user32/tests/msg.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/dlls/user32/tests/msg.c b/dlls/user32/tests/msg.c
index 11111111111..11111111111 100644
--- a/dlls/user32/tests/msg.c
+++ b/dlls/user32/tests/msg.c
@@ -12600,7 +12600,8 @@ static void test_PeekMessage(void)
     PostMessageA(info.hwnd, WM_CHAR, 'z', 0);
     qstatus = GetQueueStatus(qs_all_input);
     ok(qstatus == MAKELONG(QS_POSTMESSAGE, QS_POSTMESSAGE|QS_KEY) ||
-       qstatus == MAKELONG(QS_POSTMESSAGE, QS_POSTMESSAGE|QS_KEY|QS_SENDMESSAGE),
+       qstatus == MAKELONG(QS_POSTMESSAGE, QS_POSTMESSAGE|QS_KEY|QS_SENDMESSAGE) ||
+       broken(qstatus == MAKELONG(QS_POSTMESSAGE|QS_SENDMESSAGE, QS_POSTMESSAGE|QS_KEY|QS_SENDMESSAGE)) /* sometimes on non-us w1064v1809 */,
        "wrong qstatus %08lx\n", qstatus);
 
     InvalidateRect(info.hwnd, NULL, FALSE);
@@ -12681,6 +12682,10 @@ static void test_PeekMessage(void)
 
     msg.message = 0;
     ret = PeekMessageA(&msg, 0, 0, 0, PM_REMOVE | PM_QS_PAINT);
+    /* GetQueueStatus documentation says that it's not a guarantee that PeekMessage will succeed,
+     * it indeed fails from time to time on the non-us w1064v1809 testbot VMs, let's try again */
+    if (!ret && GetQueueStatus(qs_all_input) == MAKELONG(0, QS_PAINT|QS_KEY))
+        ret = PeekMessageA(&msg, 0, 0, 0, PM_REMOVE | PM_QS_PAINT);
     ok(ret && msg.message == WM_PAINT,
        "got %d and %04x instead of TRUE and WM_PAINT\n", ret, msg.message);
     DispatchMessageA(&msg);
-- 
2.36.1

